### PR TITLE
update python bindings

### PR DESCRIPTION
**Pull request by radhermit**
Since most of this has been sitting in my local fork for a while, it's past time to get it reviewed and merged.

The main point of this was to get some semblance of continuous mode working on the python side, which is done by using signal measure callbacks, pushing values into a queue per signal, and then combining the data from each signal into tuple pairs per sample. It would be nice if libsmu had functionality to do a callback per entire sample, but I didn't add that since I wasn't sure you wanted to head that direction with the core C++ lib.

In essence, the continuous mode is exposed via an iterator even though continuously generated data flows don't naturally lend themselves to lazy pythonic iterators. This is worked around by only queuing 1024 samples at maximum and tossing the rest while samples are not being consumed. Therefore, if an iterator is initialized and not used right away the first 1024 samples will be the oldest, original samples. The next() method of the iterator also does a busy wait until it sees matching samples across all signal queues before allowing output processing to occur.

More work should probably be done on the proto piece but I added a Device class to more easily expose the overall device sampling functionality and tried to tie it into the current layout.

Suggestions, comments, or criticism welcome (especially since my C++ is pretty rusty).

### CODE REVIEW

**Comment by itdaniher** [comment]
Looks pretty good! I'd like to see a helper method for the iterator allowing the existing data buffer to be flushed, but that could be implemented trivially in pysmu_proto.

File: pysmu.cpp [hunk 1]
```diff
@@ -3,267 +3,425 @@
...
+		PyObject* samples = PyList_New(0);
+		for (int i = 0; i < nsamples; i++) {
+			PyObject* sample_tuple = Py_BuildValue("(f,f)", buf_v[i], buf_i[i]);
+			PyList_Append(samples, sample_tuple);
+			Py_DECREF(sample_tuple);
+		}
+		return samples;
+	}
+
+	static PyObject* getAllInputs(PyObject* self, PyObject* args) {
```
Comment (itdaniher) [inline pysmu.cpp]: @radhermit it looks like this method only returns all inputs across a given device?

Comment (radhermit) [inline pysmu.cpp]: Correct, none of the python methods iterate over all session devices yet as far as I know.

**Comment by itdaniher** [comment]
@radhermit should I merge this for "better than nothing" async support or are you expecting to have multidevice functionality polished? need any additional feedback?

**Comment by itdaniher** [comment]
@radhermit Anything to commit locally?

**Comment by radhermit** [comment]
Rebased taking Kevin's previous comments into account.

**Comment by itdaniher** [comment]
Looks good! Docs, demos, and multidevice support can come next. :)

**Pull request merged**